extensions-core/namespace-lookup/src/main/java/io/druid/server/lookup/namespace/StaticMapExtractionNamespaceCacheFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.server.lookup.namespace;
+
+import io.druid.query.lookup.namespace.ExtractionNamespaceCacheFactory;
+import io.druid.query.lookup.namespace.StaticMapExtractionNamespace;
+
+import java.util.Map;
+import java.util.UUID;
+import java.util.concurrent.Callable;
+
+public class StaticMapExtractionNamespaceCacheFactory
+    implements ExtractionNamespaceCacheFactory<StaticMapExtractionNamespace>
+{
+  private final String VERSION = UUID.randomUUID().toString();
+
+  @Override
+  public Callable<String> getCachePopulator(
+      final String id,
+      final StaticMapExtractionNamespace extractionNamespace,
+      final String lastVersion,
+      final Map<String, String> swap
+  )
+  {
+    return new Callable<String>()
+    {
+      @Override
+      public String call() throws Exception
+      {
+        if (VERSION.equals(lastVersion)) {
+          return null;
+        } else {
+          swap.putAll(extractionNamespace.getMap());
```

Comment:
what do you mean?

getMap will return the map in the extraction namespace that the user had specified.